packages/svelte/src/internal/client/reactivity/sources.js

```diff
@@ -178,7 +179,13 @@ export function set(source, value) {
 				flush_sync();
 			}
 			for (const effect of inspects) {
-				update_effect(effect);
+				// If the inspect effect is clean, mark it maybe dirty so we can avoid overfiring.
```

Comment:
```suggestion
				// Mark clean inspect-effects as maybe dirty and then check their dirtiness
				// instead of just updating the effects - this way we avoid overfiring.
```